ospfd/ospf_vty.c

```diff
@@ -11177,6 +11177,70 @@ DEFUN (show_ip_ospf_vrfs,
 
 	return CMD_SUCCESS;
 }
+DEFPY (clear_ip_ospf_neighbor,
+       clear_ip_ospf_neighbor_cmd,
+       "clear ip ospf [(1-65535)]$instance neighbor [A.B.C.D$nbr_id]",
+       CLEAR_STR
+       IP_STR
+       "OSPF information\n"
+       "Instance ID\n"
+       "Reset OSPF Neighbor\n"
+       "Neighbor ID\n")
+{
+	struct listnode *node;
+	struct ospf *ospf = NULL;
+
+	/* If user does not specify the arguments,
+	 * instance = 0 and nbr_id = 0.0.0.0
+	 */
+	if (instance != 0) {
+		/* This means clear only the particular ospf process */
+		ospf = ospf_lookup_instance(instance);
+		if (ospf == NULL)
+			return CMD_NOT_MY_INSTANCE;
+	}
```

Comment:
I think you forget to reset this particular instance by a call to `ospf_neighbor_reset(ospf, nbr_id, nbrid_str)`

Reply:
No I have not forget :)
Actually, a separate process(daemon) runs for each instance, hence if the instance matches, the following for loop code will get executed and it will make sure that the particular instance is reset.
If no instance is specified then all the instances will get reset.